src/firebolt/db/connection.py

```diff
@@ -57,7 +59,7 @@ def cursor(self) -> Cursor:
     @wraps(AsyncBaseConnection._aclose)
     def close(self) -> None:
         with self._closing_lock.gen_wlock():
-            return async_to_sync(super()._aclose)()
+            return self._loop.run_until_complete(self._aclose())
```

Comment:
Would using https://docs.python.org/3/library/asyncio-eventloop.html#asyncio.get_event_loop be beneficial in here since this way if a loop is still running we don't create a new one?

Reply:
Well, I actually don't think that creating an additional loop rather than reusing an existing one would be much slower in this case. I think that here it's more just a matter of organization since we only run one task at most in parallel

Reply:
Also, the biggest point here is to be able to control when the loop is being closed

Reply:
I think we run one task at a time anyway, but I agree with the second point, better to be explicit when closing this loop.